stdlib/http/src/main/java/org/ballerinalang/net/http/websocketclientendpoint/FailoverClientInitEndpoint.java

```diff
@@ -0,0 +1,109 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specif ic language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.net.http.websocketclientendpoint;
+
+import org.ballerinalang.jvm.scheduling.Strand;
+import org.ballerinalang.jvm.values.ArrayValue;
+import org.ballerinalang.jvm.values.MapValue;
+import org.ballerinalang.jvm.values.ObjectValue;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+import org.ballerinalang.net.http.HttpConstants;
+import org.ballerinalang.net.http.HttpUtil;
+import org.ballerinalang.net.http.WebSocketConstants;
+import org.ballerinalang.net.http.exception.WebSocketException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+
+import static org.ballerinalang.net.http.WebSocketConstants.CONNECTOR_FACTORY;
+import static org.ballerinalang.net.http.WebSocketConstants.FAILOVER_CONFIG;
+import static org.ballerinalang.net.http.WebSocketConstants.RETRY_CONFIG;
+import static org.ballerinalang.net.http.WebSocketConstants.TARGET_URLS;
+import static org.ballerinalang.net.http.WebSocketUtil.getWebSocketService;
+import static org.ballerinalang.net.http.WebSocketUtil.initialiseWebSocketFailoverConnection;
+import static org.ballerinalang.net.http.WebSocketUtil.populateFailoverConnectorConfig;
+import static org.ballerinalang.net.http.WebSocketUtil.populateRetryConnectorConfig;
+
+/**
+ * Initialize the failover WebSocket Client.
+ */
+
+@BallerinaFunction(
+        orgName = WebSocketConstants.BALLERINA_ORG,
+        packageName = WebSocketConstants.PACKAGE_HTTP,
+        functionName = "init",
+        receiver = @Receiver(
+                type = TypeKind.OBJECT,
+                structType = WebSocketConstants.FAILOVER_WEBSOCKET_CLIENT,
+                structPackage = WebSocketConstants.FULL_PACKAGE_HTTP
+        )
+)
+public class FailoverClientInitEndpoint {
+
+    private static final Logger logger = LoggerFactory.getLogger(FailoverClientInitEndpoint.class);
+
+    public static void init(Strand strand, ObjectValue webSocketClient) throws URISyntaxException {
+        @SuppressWarnings(WebSocketConstants.UNCHECKED)
+        MapValue<String, Object> clientEndpointConfig = (MapValue<String, Object>) webSocketClient.getMapValue(
+                HttpConstants.CLIENT_ENDPOINT_CONFIG);
+        ArrayValue targets = clientEndpointConfig.getArrayValue(TARGET_URLS);
+        ArrayList<String> newTargetUrls = new ArrayList<>();
+        int index = 0;
+        // check whether url has valid format or not
+        // if It isn't in the valid format, remove that from the url set
+        for (int i = 0; i < targets.size(); i++) {
```

Comment:
I suppose we could use a `foreach` use instead of a `for` loop here since we are not really making use of the counter `i`.